google/cloud/storage/blob.py

```diff
@@ -2931,73 +2932,99 @@ def compose(
             (Optional) The amount of time, in seconds, to wait
             for the server response.  See: :ref:`configuring_timeouts`
 
-        :type if_generation_match: list of long
+        :type if_generation_match: long
         :param if_generation_match:
-            (Optional) Make the operation conditional on whether the blob's
-            current generation matches the given value.  Setting to 0 makes the
-            operation succeed only if there are no live versions of the blob.
+            (Optional) Makes the operation conditional on whether the
+            destination object's current generation matches the given value.
+            Setting to 0 makes the operation succeed only if there are no live
+            versions of the object.
+
+            If a list of long is passed in, makes the operation conditional on whether the
+            current generation of each source blob matches the corresponding generation.
             The list must match ``sources`` item-to-item.
+            (Deprecated: type(list of long) is supported for backwards-compatability reasons only.
+            Please use if_source_generation_match instead to match source objects generations.)
 
-        :type if_metageneration_match: list of long
+        :type if_metageneration_match: long
         :param if_metageneration_match:
-            (Optional) Make the operation conditional on whether the blob's
-            current metageneration matches the given value. The list must match
-            ``sources`` item-to-item.
+            (Optional) Makes the operation conditional on whether the
+            destination object's current metageneration matches the given
+            value.
+
+            If a list of long is passed in, no match operation will be performed.
```

Comment:
This didn't behave like if_source_metageneration_match before? It just did nothing?

Reply:
I don't think source objects metagenerations checks are performed here, according to the [API doc](https://cloud.google.com/storage/docs/json_api/v1/objects/compose#request-body).